salt/odl-video/s3_buckets_cloudfront.sls

```diff
@@ -0,0 +1,112 @@
+  boto_s3_bucket.present:
+    - Bucket: {{ bucket_prefix}}-{{ bucket_purpose }}-{{ bucket_suffix }}
+    - region: us-east-1
+    - CORSRules:
+      - AllowedOrigin: ["*"]
+        AllowedMethod: ["GET"]
+        AllowedHeader: ["Authorization"]
+        MaxAgeSconds: 3000
+    - Versioning:
+        Status: "Enabled"
+{% endfor %}
+{% endfor %}
+
+{% for odl_video_bucket_suffix in ['rc', 'prod'] %}
+create_cloudfront_distribution_{{ odl_video_bucket_prefix }}-{{ odl_video_bucket_suffix }}:
+  boto_cloudfront.present:
+  - name: {{ odl_video_bucket_prefix }}-{{ odl_video_bucket_suffix }}
+  - config:
+      CacheBehaviors:
+        Items:
+        - AllowedMethods:
+            CachedMethods:
+              Items:
+              - HEAD
+              - GET
+              - OPTIONS
+            Items:
+            - HEAD
+            - GET
+            - OPTIONS
+          Compress: false
+          DefaultTTL: 86400
+          ForwardedValues:
+            Cookies:
+              Forward: none
+            Headers:
+              Items:
+              - Access-Control-Request-Headers
+              - Access-Control-Request-Method
+              - Origin
+            QueryString: false
+          MaxTTL: 31536000
+          MinTTL: 0
+          PathPattern: /{{ odl_video_bucket_purposes }}-{{ odl_video_bucket_suffix }}*
+          SmoothStreaming: false
+          TargetOriginId: S3-{{ odl_video_bucket_prefix }}-{{ odl_video_bucket_purposes }}-{{ odl_video_bucket_suffix }}
+          TrustedSigners:
+            Enabled: false
+          ViewerProtocolPolicy: redirect-to-https
+      DefaultCacheBehavior:
+        AllowedMethods:
+          CachedMethods:
+            Items:
+            - HEAD
+            - GET
+            - OPTIONS
+          Items:
+          - HEAD
+          - GET
+          - OPTIONS
+        Compress: false
+        DefaultTTL: 86400
+        ForwardedValues:
+          Cookies:
+            Forward: none
+          Headers:
+            Items:
+            - Access-Control-Request-Headers
+            - Access-Control-Request-Method
+            - Origin
+          QueryString: false
+        MaxTTL: 31536000
+        MinTTL: 0
+        SmoothStreaming: false
+        TargetOriginId: S3-{{ odl_video_bucket_prefix }}-{{ odl_video_bucket_suffix }}
+        TrustedSigners:
+          Enabled: true
+          Items:
+          - self
+        ViewerProtocolPolicy: redirect-to-https
+      DefaultRootObject: ''
+      Enabled: true
+      HttpVersion: http2
+      IsIPV6Enabled: true
+      Logging:
+        Bucket: ''
+        Enabled: false
+        IncludeCookies: false
+        Prefix: ''
+      Origins:
+        Items:
+          DomainName: {{ odl_video_bucket_prefix }}-{{ odl_video_bucket_purposes }}-{{ odl_video_bucket_suffix }}.s3.amazonaws.com
+          Id: S3-{{ odl_video_bucket_prefix }}-{{ odl_video_bucket_purposes }}-{{ odl_video_bucket_suffix }}
```

Comment:
Same issue here with rendering the full list of purposes.

Reply:
Changed by adding an additional for loop